frontend/src/metabase/entities/users/forms.js

```diff
@@ -84,20 +84,22 @@ export default {
       ...PLUGIN_ADMIN_USER_FORM_FIELDS,
     ],
   },
-  user: {
-    fields: user => {
-      const isSsoUser = !PLUGIN_IS_PASSWORD_USER.every(predicate =>
-        predicate(user),
-      );
-
-      if (isSsoUser) {
-        return [getLocaleField()];
-      }
-
+  user: user => {
```

Comment:
Reorganize how user form is constructed since the `fields()` function doesn't play nice with passed `user`. More detail in this comment.